package-lock.json

```diff
@@ -0,0 +1,13896 @@
+{
+  "name": "cortex",
```

Comment:
Just my personal gripe, generally speaking I really don't get the point of committing package-lock. I know npm says you should and all, but its ordering seems pretty non-deterministic so if anything changes when you build, you have to commit a randomly changed package-lock. It seems pointless for rebuilding stuff, since package.json contains all the info you need, so why bother having this messy file in version control?
I'm not saying you should remove it from git or anything, I'm just curious about your thoughts on this. I've never had anyone really justify it to me in conversations I've had in the past.

Reply:
In this case I think the main advantage of having it checked in is that it enables security scans on dependencies.
https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/about-alerts-for-vulnerable-dependencies#detection-of-vulnerable-dependencies

The package-lock.json defines the full dependency tree, just the package.json is not enough for this sort of check.

Generally I think the reason it's treated as a best practice is that it allows for a deterministic release build. If you are building off of the package.json alone, you might see the same source code fail on successive builds.

Reply:
Can't/doesn't npm tell you about vulnerabilities in your plugins already? Fair point though regarding security scans, having the full tree is obviously needed for that to work.
And okay, I don't think I've ever worked with a system that used it to build deterministically (or if it did, it did it quietly under the hood and I never noticed). I do see the honest benefit of that then.

Thanks for that info haha.